### PR TITLE
fix: enable pool error handling

### PR DESCRIPTION
**Pull request by mgabeler-lee-6rs**
As documented in https://node-postgres.com/api/pool#events, it is _very_ important to attach a listener to the Pg Pool `error` event if you don't want your app to randomly crash/exit if something happens to idle connections in the pool (e.g. someone restarts your PostgreSQL server/pgBouncer/etc., or server times out your idle connection). Currently `loopback-connector-postgresql` provides no means of doing this, which makes writing stable apps that use it hard.

This PR adds an optional `onError` property to the connector settings, which can be one of:
* The string `ignore` to report any errors via `debug`, and otherwise ignore them
* An app-provided listener function to connect to the pool

It also, as a side item, fixes a broken link to the node-postgres documentation.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

*Testing notes:* The tests here verify it wires up the error handler the way it is documented to work. Still WIP on verifying this in a real app to ensure it really addresses the stability issues.

### CODE REVIEW

**Comment by dhmlau** [comment]
@slnode test please

**Review by dhmlau — APPROVED** [review]
@mgabeler-lee-6rs, thanks for your PR. Your changes look reasonable to me. I'd like to get at least one more review/approval before landing it. Thanks.

**Comment by dhmlau** [comment]
Commit linter and PR linter checks seem to be stuck. Kicking off CI again.
@slnode test please

**Comment by dhmlau** [comment]
@rmg, the commit linter and PR linter is stuck. I tried to kick off CI twice with same results. Could you please help? Thanks!

**Review by bajtos — APPROVED** [review]
LGTM, thank you @mgabeler-lee-6rs for the contribution 👏🏻

**Comment by mgabeler-lee-6rs** [comment]
> Testing notes: The tests here verify it wires up the error handler the way it is documented to work. Still WIP on verifying this in a real app to ensure it really addresses the stability issues.

For posterity: have confirmed I can wire up this hook in my applications and, with that in place, restarting the postgresql server no longer crashes the app

**Pull request merged**
